.github/workflows/pr-validation.yml

```
@@ -65,4 +68,75 @@ jobs:
+      - name: Get changed files
+        id: changed-files
+        uses: tj-actions/changed-files@v46
+        with:
+          files: |
+            **/*.go
+          files_ignore: |
+            **/*_test.go
+
+      - name: Test changed packages
+        if: steps.changed-files.outputs.any_changed == 'true'
+        run: |
```

actionlint (1.7.7)

shellcheck reported issue in this script: SC2086:info:11:65: Double quote to prevent globbing and word splitting

(shellcheck)